filebeat/input/systemlogs/input.go

```diff
@@ -0,0 +1,234 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package systemlogs
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/elastic/beats/v7/filebeat/channel"
+	v1 "github.com/elastic/beats/v7/filebeat/input"
+	loginput "github.com/elastic/beats/v7/filebeat/input/log"
+	v2 "github.com/elastic/beats/v7/filebeat/input/v2"
+	cursor "github.com/elastic/beats/v7/filebeat/input/v2/input-cursor"
+	"github.com/elastic/beats/v7/libbeat/feature"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+const pluginName = "system-logs"
+
+func init() {
+	// Register an input V1, that's used by the log input
+	if err := v1.Register(pluginName, newV1Input); err != nil {
+		panic(err)
+	}
+}
+
+type config struct {
+	UseJournald bool    `config:"use_journald"`
+	UseFiles    bool    `config:"use_files"`
+	Files       *conf.C `config:"files" yaml:"files"`
+	Journald    *conf.C `config:"journald" yaml:"journald"`
+}
+
+func (c *config) Validate() error {
+	if c.UseFiles && c.UseJournald {
+		return errors.New("'use_journald' and 'use_files' cannot both be true")
+	}
+
+	if c.Files == nil && c.Journald == nil {
+		return errors.New("one of 'journald' or 'files' must be set")
+	}
+
+	return nil
+}
+
+// newV1Input checks whether the log input must be created and
+// delegates to loginput.NewInput if needed.
+func newV1Input(
```

Comment:
It has integration tests, the python system tests.

But I can write some unit/integration (integration in the sense they will rely on `journalclt` existing on the host) if you prefer.

Reply:
I would prefer if there were tests that lived with the input, right now they live with the system module which is quite far away from the input. In a year from now someone reading this may have trouble spotting them.

What I see now is this configuration in:

```json
        if ".journal" in test_file:
            cmd.remove("-once")
            cmd.append("-M")
            cmd.append("{module}.{fileset}.var.use_journald=true".format(
                module=module, fileset=fileset))
            cmd.append("-M")
            cmd.append("{module}.{fileset}.input.journald.paths=[{test_file}]".format(
                module=module, fileset=fileset, test_file=test_file))
        else:
            cmd.append("-M")
            cmd.append("{module}.{fileset}.var.paths=[{test_file}]".format(
                module=module, fileset=fileset, test_file=test_file))
```

This tests the `use_journald` configuration and also the case where `paths` exists. These are not exhaustive tests :)

I don't see a test for the case where there is a list of file paths, but the paths don't exist in the file system, so we read from journald. This is the most common case we expect to it.

Reply:
I'll add some tests for this case on a new PR.

Reply:
I created a GH issue to help keep track of it: https://github.com/elastic/beats/issues/41259